website/blog/2023-02-21-ferretdb-compare-arrays.md

```diff
@@ -0,0 +1,380 @@
+`$lt` selects the smallest element from a nested array, while `$gt` compares index to index.
+
+We are yet to fully understand how a nested array is expected to work with a filter comparison operator.
+
+**Example:** `$gt` Operator with Nested Array Argument
+
+:::caution
+This example also shows an uncertain result of the `$gt` operator.
+:::
+
+The following example shows the `$gt` operator with a nested array argument `[['c']]`.
+
+```js
+db.gtexample.find({v: {$gt: [['c']]}})
+```
+
+The output:
+
+```js
+[
+  { _id: 'array-bool', v: [ false ] },
+  { _id: 'nested-mixed', v: [ [ 'c' ], [ 1, 'z', 'b' ] ] },
+  { _id: 'nested-bool', v: [ [ true ] ] }
+]
+```
+
+No scalar values from the collection were selected since the scalar values are not array type.
+
+The selected array `array-bool` contains a boolean value, which has a greater BSON comparison order than that of an array type.
+Therefore, when comparing `['c']` with the items in the array, the largest item from the array (`false`) was used for the comparison.
+
+Using the BSON comparison order, the boolean value is considered greater than an array, hence the entire `array-bool` array was selected.
+
+When dealing with nested arrays, the arrays `nested-mixed` and `nested-bool` were selected when using the `$gt` operator on an array filter argument.
+
+In `$gt`, index-to-index comparison is performed between the filter argument and the arrays in the collection.
+
+For `nested-mixed`, both the array in the collection and the filter argument have the same element `['c']` at the first index, but the array in the collection has an additional item at the second index, making it greater than the filter argument.
+
+For `nested-bool`, the array in the collection has a boolean value at the first index of the inner array, which has a greater BSON comparison order than the string in the filter argument.
+This makes `nested-bool` greater than the filter argument.
+
+Unlike `$lt` which takes the smallest element from the array, nested array is compared index to index in `$gt`.
+
+## Nested array at FerretDB
+
+At FerretDB, we do not support nested arrays at the moment.
+This is due to the complexities caused by
+differences in the comparison of array and other BSON types.
+
+In particular, differentiating between the cases where the BSON comparison order of an array is used versus an element from the comparison array was not entirely straightforward.
+
+Since then, we’ve taken time to understand the inner workings of comparison for various operations.
+
+After investigating the behaviour of the `$lt` and `$gt` operators with array and nested array filter arguments, we discovered that the comparison process is different between the two cases.
+
+However, given an array argument filter and nested array collection, `$lt` uses the smallest element from the nested array and `$gt` on the other hand compares index to index of nested array collection.
```

Comment:
```suggestion
However, given an array argument filter and array collection, `$lt` uses the smallest element from the array and `$gt` on the other hand compares index to index of array collection.
```
It also works like that for nested array as well. But important to mention that array along is difficult